### PR TITLE
[DOCS]Add Catalog Promotion documentation

### PR DESCRIPTION
**Pull request by arti0090**
| Q               | A
| --------------- | -----
| Branch?         |  master 
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| License         | MIT

### CODE REVIEW

**Review by AdamKasp — APPROVED** [review]

File: docs/book/products/catalog_promotions.rst [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+.. code-block:: bash
+
+    {
+        #...
+        "rules": [
+            {
+                "type": "for_variants",
+                "configuration": {
+                    "variants": [
+                        "Everyday_white_basic_T_Shirt-variant-1", #Variant Code
```
[the remark this answers is not in the document]
Comment (arti0090) [reply docs/book/products/catalog_promotions.rst:59]: Do you have something in mind?

File: docs/book/products/catalog_promotions.rst [hunk 2]
```diff
@@ -0,0 +1,276 @@
+.. index::
+   single: Catalog Promotions
+
+Catalog Promotions
+==================
+
+The **Catalog Promotions** system in **Sylius** is a new way of dealing with promotions on multiple products.
+If you get used to `Cart Promotions </book/orders/cart-promotions>` this will be something familiar to you.
+
+It is managed by combination of promotion rules and actions, where you can specify on which f.e. products or taxons
```
Comment (GSadee) [inline docs/book/products/catalog_promotions.rst:10]:
```suggestion
It is managed by combination of promotion rules and actions, where you can specify on which e.g. products or taxons
```

File: docs/book/products/catalog_promotions.rst [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+you can specify the Catalog Promotion with your custom actions as well as actions like percentage or value discount.
+
+You can assign the needed channels too.
+
+Catalog Promotion Parameters
+----------------------------
+
+Catalog Promotion has a few basic parameters that represents it - a unique ``code`` and ``name``:
```
Comment (GSadee) [inline docs/book/products/catalog_promotions.rst:18]:
```suggestion
Catalog Promotion has a few basic parameters that represent it - a unique ``code`` and ``name``:
```

File: docs/book/products/catalog_promotions.rst [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+      'https://hostname/api/v2/admin/catalog-promotions' \
+      -H 'accept: application/ld+json' \
+      -H 'Authorization: Bearer authorizationToken' \
+      -H 'Content-Type: application/ld+json' \
+      -d '{
+        "code": "t_shirt_promotion",
+        "name": "T-shirt Promotion"
+        }'
+
+
```
Comment (GSadee) [inline docs/book/products/catalog_promotions.rst:129]:
```suggestion
```

File: docs/book/products/catalog_promotions.rst [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+    $catalogPromotionRule = $this->catalogPromotionRuleExampleFactory->create($rule);
+    $catalogPromotionRule->setCatalogPromotion($catalogPromotion);
+    $catalogPromotion->addRule($catalogPromotionRule);
+
+    /** @var CatalogPromotionActionInterface $catalogPromotionAction */
+    $catalogPromotionAction = $this->catalogPromotionActionExampleFactory->create($action);
+    $catalogPromotionAction->setCatalogPromotion($catalogPromotion);
+    $catalogPromotion->addAction($catalogPromotionAction);
+
+And now you should be able to see created Catalog Promotion. You can check if it exist like in last example (with GET endpoint).
```
Comment (GSadee) [inline docs/book/products/catalog_promotions.rst:228]:
```suggestion
And now you should be able to see created Catalog Promotion. You can check if it exists like in the last example (with GET endpoint).
```

File: docs/book/products/catalog_promotions.rst [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+
+.. code-block:: bash
+
+    curl -X 'GET' \
+    'https://hostname/api/v2/shop/product-variant/Everyday_white_basic_T_Shirt-variant-1'
+
+.. code-block:: bash
+
+    # response content
+
```
Comment (GSadee) [inline docs/book/products/catalog_promotions.rst:239]:
```suggestion
```

**Review by GSadee — APPROVED** [review]

File: docs/book/products/catalog_promotions.rst [hunk 7]
```diff
@@ -0,0 +1,276 @@
...
+   single: Catalog Promotions
+
+Catalog Promotions
+==================
+
+The **Catalog Promotions** system in **Sylius** is a new way of dealing with promotions on multiple products.
+If you get used to `Cart Promotions </book/orders/cart-promotions>` this will be something familiar to you.
+
+It is managed by combination of promotion rules and actions, where you can specify on which f.e. products or taxons
+you can specify the Catalog Promotion with your custom actions as well as actions like percentage or value discount.
```
Comment (AdamKasp) [inline docs/book/products/catalog_promotions.rst:11]:
```suggestion
you can specify the Catalog Promotion with your custom actions as well as actions like percentage discount.
```
it is not true yet.

**Review by AdamKasp — APPROVED** [review]

**Review by SirDomin — APPROVED** [review]

**Comment by SirDomin** [comment]
Thank you, @arti0090! :tada:

**Pull request merged**